nexus/src/app/saga.rs

```diff
@@ -149,14 +150,27 @@ impl super::Nexus {
 
         let result = runnable_saga.fut.await;
         result.kind.map_err(|saga_error| {
-            saga_error
+            let mut error = saga_error
                 .error_source
                 .convert::<Error>()
                 .unwrap_or_else(|e| Error::internal_error(&e.to_string()))
                 .internal_context(format!(
-                    "saga error at node {:?}",
+                    "saga ACTION error at node {:?}",
                     saga_error.error_node_name
-                ))
+                ));
+            if let Some((undo_node, undo_error)) = saga_error.undo_failure {
+                error = error.internal_context(format!(
+                    "UNDO ACTION failed (node {:?}, error {:#}) after",
+                    undo_node, undo_error
+                ));
+
+                error!(log, "saga stuck";
+                    "saga_id" => runnable_saga.id.to_string(),
+                    "error" => format!("{:#}", e),
```

Comment:
Nit, but I believe you can format the error directly, without converting to a string. Something like `"error" => #%e`, I think. (From [here](https://docs.rs/slog/latest/slog/macro.log.html#fmtdisplay-and-fmtdebug-values).)

Reply:
Cool.  Will try that.